virttest/qemu_devices/qcontainer.py

```diff
@@ -2904,20 +2904,27 @@ def dimm_device_define_by_params(self, params, name):
         """
         Create pc-dimm device from params.
         """
-        params = params.object_params("dimm")
+        # Ensure get accurate params for dimm device, avoid irrelevant params
+        # be used by mistake.
+        dimm_params = Params()
+        suffix = '_dimm'
+        for key in list(params.keys()):
+            if key.endswith(suffix):
+                new_key = key.rsplit(suffix)[0]
+                dimm_params[new_key] = params[key]
+
         dimm_type = "nvdimm" if params.get("nv_backend") else "pc-dimm"
-        attrs = qdevices.Dimm.__attributes__[:]
-        dev = qdevices.Dimm(params=params.copy_from_keys(attrs),
-                            dimm_type=dimm_type)
-        dev.set_param("id", "%s-%s" % ("dimm", name))
-        if dimm_type == "nvdimm" and params.get("nvdimm_uuid"):
+        attrs = qdevices.Dimm.__attributes__[dimm_type][:]
+        dimm_uuid = dimm_params.get("uuid")
+        if 'uuid' in attrs and dimm_uuid:
             try:
-                dev.set_param("uuid", uuid.UUID(params["nvdimm_uuid"]))
+                dimm_params['uuid'] = uuid.UUID(dimm_uuid)
```

Comment:
```suggestion
                dimm_params['uuid'] = str(uuid.UUID(dimm_uuid))
```

Reply:
Make sense, we do not need a class.  Updated, thanks!